nupic/algorithms/CLAClassifier.py

```diff
@@ -558,13 +560,13 @@ def read(cls, proto):
     classifier._patternNZHistory = deque(maxlen=max(classifier.steps) + 1)
     patternNZHistoryProto = proto.patternNZHistory
     learnIteration = classifier._learnIteration - len(patternNZHistoryProto) + 1
-    for i in xrange(len(patternNZHistoryProto)):
+    for i in range(len(patternNZHistoryProto)):
```

Comment:
This is an example of an automatic change that is inappropriate in most circumstances for python 2.x:

1. it adds unnecessary overhead of creating a list on python 2.x
2. In the cases where the parameter to `xrange` was a really large number, this would generate a huge list on python 2.x